pkg/controller/node/controller_utils.go

```diff
@@ -56,45 +57,59 @@ func cleanupOrphanedPods(pods []*api.Pod, nodeStore cache.Store, forcefulDeleteP
 	}
 }
 
-// deletePods will delete all pods from master running on given node, and return true
-// if any pods were deleted, or were found pending deletion.
-func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, nodeName, nodeUID string, daemonStore cache.StoreToDaemonSetLister) (bool, error) {
-	remaining := false
-	selector := fields.OneTermEqualSelector(api.PodHostField, nodeName)
-	options := api.ListOptions{FieldSelector: selector}
-	pods, err := kubeClient.Core().Pods(api.NamespaceAll).List(options)
-	if err != nil {
-		return remaining, err
+func tolerationsToleratesTaints(tolerations []api.Toleration, taints []api.Taint) bool {
```

Comment:
Sure, i will reduce duplication. I concluded that scheduler is a separate package (plugin) and it is undesirable to add cross-dependencies between core and scheduler.